web/src/app/services/auth/passlist.guard.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2025 The Ground Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Injectable} from '@angular/core';
+
+import {AuthService} from 'app/services/auth/auth.service';
+import {NavigationService} from 'app/services/navigation/navigation.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PasslistGuard {
+  constructor(
+    private authService: AuthService,
+    private navigationService: NavigationService
+  ) {}
+
+  async canActivate(): Promise<void> {
```

Comment:
"activate" is more of an internal term that I think we'd want here... Also, this does more than just check whether the user can activate the survey, it actually redirects to the form. Perhaps something like "checkSurveyAcls" might be more appropriate?